### PR TITLE
Add NetApp SolidFire volume access group module

### PR DESCRIPTION
**Pull request by timuster**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

NetApp SolidFire volume access group module

##### ANSIBLE VERSION

```
ansible 2.3.0
  config file = 
  configured module search path = Default w/o overrides
```

##### SUMMARY

Adding sf_volume_access_group_manager.py, a module to manage NetApp SolidFire volume access groups. The design was based on the existing NetApp E-Series modules.

```

```

### CODE REVIEW

**Comment by hulquest** [comment]
All of the general comments made in #20732 have been applied here.  
The module is sound from a SF API point of view. 
shipit

File: lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+module: sf_volume_access_group_manager
+
+short_description: Manage SolidFire Volume Access Groups
+extends_documentation_fragment:
+    - netapp.solidfire
+version_added: '2.3'
+author: Sumit Kumar (sumit4@netapp.com)
+description:
+- Create, destroy, or update volume access groups on SolidFire
+    - auth_basic
```
Comment (resmo) [inline lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py]: is this a left over?

File: lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+options:
+
+    state:
+        required: true
```
Comment (resmo) [inline lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py]: many modules have the default of `state=present`. consider this as a default if you feel it make sense.

File: lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+
+EXAMPLES = """
+   - name: Create Volume Access Group
+     sf_volume_access_group_manager:
+       hostname: "{{ solidfire_hostname }}"
+       username: "{{ solidfire_username }}"
+       password: "{{ solidfire_password }}"
+       state: present
+       name: AnsibleVolumeAccessGroup
+       volumes: 7,8
```
Comment (resmo) [inline lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py]: I would prefer the syntax `[7, 8]` comma separated can be confusing

File: lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+    type: string
+    sample: '{"changed": true, "key": value}'
+
+msg:
+    description: Successful update of Volume Access Group
+    returned: success
+    type: string
+    sample: '{"changed": true}'
+
+msg:
```
Comment (resmo) [inline lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py]: changed: true is a common return, no need to document that. but you have `msg` several times, which does not make much sense. if the module does not return anything other than the common returns, just leave it empty

File: lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+        default: None
+
+    virtual_network_tags:
+        required: false
+        type: int[]
+        description:
+        - The ID of the VLAN Virtual Network Tag to associate the volume access group with.
+        default: None
+
+    attributes:
```
Comment (resmo) [inline lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py]: would be good to have an example for this as it is free form

Comment (timuster) [reply lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py]: I've provided an example in the "EXAMPLES" section.

File: lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py (hunk 2 above)
Comment (timuster) [reply lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py]: That's a good suggestion! However, for the sake of consistency with the other existing NetApp modules, I think it's a good idea to leave it as it is. We can update all of our modules later.

**Comment by timuster** [comment]
ready_for_review

File: lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+DOCUMENTATION = '''
```
Comment (mattclay) [inline lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py]: CI failure due to missing module metadata:

```
2017-02-06 20:39:27 ============================================================================
2017-02-06 20:39:27 lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py
2017-02-06 20:39:27 ============================================================================
2017-02-06 20:39:27 ERROR: No ANSIBLE_METADATA provided
```

**Comment by resmo** [comment]
shipit

(note for the merger, add metadata after the merge)

**Comment by resmo** [comment]
shipit

**Comment by resmo** [comment]
shipit

**Comment by hulquest** [comment]
shipit

File: lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+       volume_access_group_id: 1
+"""
+
+RETURN = """
+
+
+"""
+
+
+import logging
```
Comment (resmo) [inline lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py]: please remove logging due security concerns

File: lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+
+def main():
+    v = SolidFireVolumeAccessGroup()
+
+    try:
+        v.apply()
+    except:
+        err = get_exception()
+        logger.debug("Exception in apply(): \n%s" % format_exc(err))
+        raise
```
Comment (resmo) [inline lib/ansible/modules/storage/netapp/sf_volume_access_group_manager.py]: replace with fail_json() to show a user friendly error message

**Review by resmo — CHANGES_REQUESTED** [review]

**Comment by timuster** [comment]
ready_for_review

**Review by resmo — APPROVED** [review]

**Pull request merged**